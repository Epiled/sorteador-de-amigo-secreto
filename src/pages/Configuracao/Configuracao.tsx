import Card from "../../components/Card";
import Formulario from "../../components/Formulario/Formulario";
import ListaParticipantes from "../../components/ListaParticipantes/ListaParticipantes";
import Rodape from "../../components/Rodape/Rodape";
import Titulo from "../../components/Titulo";

const Configuracao: React.FC = () => {
  return (
    <Card>
      <Titulo>Vamos come√ßar!</Titulo>
      <Formulario />
      <ListaParticipantes />
      <Rodape />
    </Card>
  );
}

export default Configuracao;