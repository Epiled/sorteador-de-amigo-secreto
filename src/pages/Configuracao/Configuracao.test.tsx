import { render } from '@testing-library/react';
import React from 'react';
import { RecoilRoot } from 'recoil';
import Configuracao from './Configuracao';

const mockNavegacao = jest.fn()

jest.mock('react-router-dom', () => {
  return {
    useNavigate: () => mockNavegacao
  }
})

describe('a pagina de configuracao', () => {

  test('deve ser renderizado corretamente', () => {
    const { container } = render(
      <RecoilRoot>
        <Configuracao />
      </RecoilRoot>
    )

    //snapshot
    expect(container).toMatchSnapshot()
  })
  
})