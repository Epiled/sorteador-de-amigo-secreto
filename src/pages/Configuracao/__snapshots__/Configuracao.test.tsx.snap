// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a pagina de configuracao deve ser renderizado corretamente 1`] = `
<div>
  <div
    class="sc-bdDqHD dFomCO"
  >
    <div
      class="sc-gtcySc gVSWal"
    >
      <h1
        class="sc-crPhqv cOgWik"
      >
        Vamos come√ßar!
      </h1>
      <form
        class="sc-hKVqoC bQoOkO"
      >
        <div
          class="sc-eCQicT fqsHFG"
        >
          <svg
            aria-hidden="true"
            class="StyledIconBase-sc-ea9ulj-0 gqstqG sc-gKkibJ bPPRiP"
            color="var(--darkColor-30)"
            fill="currentColor"
            focusable="false"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm-9-2V7H4v3H1v2h3v3h2v-3h3v-2H6zm9 4c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
          <input
            class="sc-jSpqns chwRwI"
            placeholder="Insira o nome dos participantes"
            type="text"
            value=""
          />
        </div>
        <button
          class="sc-dkXqGt dshXGx"
          disabled=""
        >
          Adicionar
        </button>
        
      </form>
      <ul
        class="sc-fuzrqj kWhhfn"
      />
      <footer
        class="sc-jrICMi cNwccD"
      >
        <button
          class="sc-dkXqGt kBpyYS"
          disabled=""
        >
          <svg
            aria-hidden="true"
            class="StyledIconBase-sc-ea9ulj-0 gqstqG sc-iqkjvU kqqEJz"
            color="var(--brightestColor)"
            fill="currentColor"
            focusable="false"
            height="40"
            style="margin-right: 25px;"
            viewBox="0 0 24 24"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m10 16.5 6-4.5-6-4.5v9zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
            />
          </svg>
          Iniciar brincadeira
        </button>
        <img
          alt="#"
          class="sc-kEbfcl gtSdmk"
          src="sacolas.png"
        />
      </footer>
    </div>
  </div>
</div>
`;
